### PR TITLE
[x] split out lint engine into its own crate

### PR DESCRIPTION
**Pull request by sunshowers**
Split up the x crate.

### CODE REVIEW

File: devtools/x/src/utils.rs [hunk 1]
```diff
@@ -5,10 +5,13 @@ use crate::{cargo::Cargo, Result};
 use serde::Deserialize;
 use std::path::{Path, PathBuf};
 
+/// The number of directories between the project root and the root of this crate.
+pub const X_DEPTH: usize = 2;
```
Comment (rexhoffman) [inline devtools/x/src/utils.rs:9]: Looks like this is the only non mechanical change.

**Comment by sunshowers** [comment]
> Just a thought on the linting context(s). Not sure if you made the context relational or not, or if they can reuse AST data from prior linters. But perhaps something to think about - didn't drill heavily in to the code just noticed the contexts, and that they are not necessarily related to eachother.

Yeah it would be fantastic to be able to express a DAG of lints that way.

**Pull request closed without merging**